src/main/java/org/stellar/sdk/Transaction.java

```diff
@@ -90,10 +87,21 @@ public Memo getMemo() {
   }
   
   /**
-   * @return TimeBounds, or null (representing no time restrictions)
+   * @return Preconditions
+   */
+  public PreconditionsV2 getPreconditions() {
+    return preconditions;
+  }
+
+  /**
+   * @return TimeBounds, or null (representing no time restrictions, i.e. infinite)
    */
   public TimeBounds getTimeBounds() {
-    return mTimeBounds;
+    TimeBounds timeBounds = TimeBounds.fromXdr(preconditions.getTimeBounds());
+    if (timeBounds == null || (timeBounds.getMinTime() < 1 && timeBounds.getMaxTime() < 1)) {
```

Comment:
This is a breaking behavioral change, right? What's the purpose of introducing the "return null if infinite timebounds"?

Reply:
oh, I removed that, it enabled  `SEP10ChallengeTest.testReadChallengeTransactionInvalidNoTimeBounds()`  to pass that infinite range on the updated tx builder in this version, and `SEP10Challenge.java` was checking for null timebounds on transaction, but that was only possible from test earlier when it used the Transaction constructor directly to pass timebounds=null, and now the builder doesn't allow that, I deleted the reference in `SEP10Challenge.java` and related test case also, as it's obsoleted by the tx builder enforcing presence of timebounds.